### PR TITLE
Fix bug that overwrites records

### PR DESCRIPTION
**Pull request by dianadoherty**
## App
```
    database = app.resource(name: "pg")

    records = database.records(collection: "sequences")

    processed_records = app.process(records: records, process: Passthrough.new)

    database.write(records: processed_records, collection: "testing1_fn_rb")

    database.write(records: records, collection: "testing1_no_fn_rb")
```

### Before
```
meroxa apps run
+------------------------------------------------+
| Destination pg/testing1_no_fn_rb               |
+-----------------+------------------------------+
|           INDEX | RECORD                       |
+-----------------+------------------------------+
|               0 | {"id":1,"message":"goodbye"} |
+-----------------+------------------------------+
| RECORDS WRITTEN | 1                            |
+-----------------+------------------------------+
+------------------------------------------------+
| Destination pg/testing1_fn_rb                  |
+-----------------+------------------------------+
|           INDEX | RECORD                       |
+-----------------+------------------------------+
|               0 | {"id":1,"message":"goodbye"} |
+-----------------+------------------------------+
| RECORDS WRITTEN | 1                            |
+-----------------+------------------------------+
```

### After
```
+----------------------------+
| Destination pg/testing1_no |
| _fn_rb                     |
+-----------------+----------+
|           INDEX | RECORD   |
+-----------------+----------+
|               0 | {"id":1} |
+-----------------+----------+
| RECORDS WRITTEN | 1        |
+-----------------+----------+
+------------------------------------------------+
| Destination pg/testing1_fn_rb                  |
+-----------------+------------------------------+
|           INDEX | RECORD                       |
+-----------------+------------------------------+
|               0 | {"id":1,"message":"goodbye"} |
+-----------------+------------------------------+
| RECORDS WRITTEN | 1                            |
+-----------------+------------------------------+
```

### CODE REVIEW

**Review by jmar910 — APPROVED** [review]

**Review by lyuboxa — APPROVED** [review]

File: lib/ruby/turbine_rb/lib/turbine_rb/client.rb [hunk 1]
```diff
@@ -23,13 +23,14 @@ def resource(name:)
       end
 
       def process(records:, process:)
+        newrcords = records.dup
```
Comment (raulb) [inline lib/ruby/turbine_rb/lib/turbine_rb/client.rb:26]: Just a typo:

```suggestion
        new_records = records.dup
```

File: lib/ruby/turbine_rb/lib/turbine_rb/client.rb [hunk 2]
```diff
@@ -23,13 +23,14 @@ def resource(name:)
...
         pb_collection = core_server.add_process_to_collection(
           TurbineCore::ProcessCollectionRequest.new(
-            collection: Collection.unwrap(records),
+            collection: Collection.unwrap(newrcords),
```
Comment (raulb) [inline lib/ruby/turbine_rb/lib/turbine_rb/client.rb:29]:
```suggestion
            collection: Collection.unwrap(new_records),
```

File: lib/ruby/turbine_rb/lib/turbine_rb/client.rb [hunk 3]
```diff
@@ -23,13 +23,14 @@ def resource(name:)
...
             process: TurbineCore::ProcessCollectionRequest::Process.new(name: process.class.name)
           )
         )
-        records.tap do |r|
+        newrcords.tap do |r|
```
Comment (raulb) [inline lib/ruby/turbine_rb/lib/turbine_rb/client.rb:33]:
```suggestion
        new_records.tap do |r|
```

File: lib/ruby/turbine_rb/lib/turbine_rb/client.rb [hunk 4]
```diff
@@ -74,13 +75,14 @@ def records(collection:, configs: nil)
         end
 
         def write(records:, collection:, configs: nil)
+          newrcords = records.dup
```
Comment (raulb) [inline lib/ruby/turbine_rb/lib/turbine_rb/client.rb:78]:
```suggestion
          new_records = records.dup
```

File: lib/ruby/turbine_rb/lib/turbine_rb/client.rb [hunk 5]
```diff
@@ -74,13 +75,14 @@ def records(collection:, configs: nil)
...
           if records.instance_of?(Collection) # it has been processed by a function, so unwrap back to gRPC collection
-            records = records.unwrap
+            newrcords = records.unwrap
```
Comment (raulb) [inline lib/ruby/turbine_rb/lib/turbine_rb/client.rb:80]:
```suggestion
            new_records = records.unwrap
```

File: lib/ruby/turbine_rb/lib/turbine_rb/client.rb [hunk 6]
```diff
@@ -74,13 +75,14 @@ def records(collection:, configs: nil)
...
           end
 
           req = TurbineCore::WriteCollectionRequest.new(
             resource: @pb_resource,
-            sourceCollection: records,
+            sourceCollection: newrcords,
```
Comment (raulb) [inline lib/ruby/turbine_rb/lib/turbine_rb/client.rb:85]:
```suggestion
            sourceCollection: new_records,
```

**Review by raulb — CHANGES_REQUESTED** [review]
@dianadoherty nice one! Only made a suggestion to fix a typo, and maybe using a variable name more Ruby like.

**Pull request merged**